docs: add info about ESM into migration guide. (#10903)
chartjs/Chart.js: docs/migration/v4-migration.md

* The order of the `ChartMeta` parameters have been changed from `<Element, DatasetElement, Type>` to `<Type, Element, DatasetElement>`.

### General

> * Chart.js becomes a [ESM-only package](https://nodejs.org/api/esm.html). To use Chart.js, your project should also be a ES module. Make sure to have this in your `package.json`:

Reviewer: [nit] "a ES module" should be "an ESM module"

Reviewer 2: ESM = ECMAScript Module, so "ESM module" = "ECMAScript Module module"? :)

Author: @kurkle there are "**ESM**" and "**ES** module"

> * Chart.js becomes an [ESM-only package](https://nodejs.org/api/esm.html). To use Chart.js, your project should also be an ES module. Make sure to have this in your `package.json`:

Reviewer 3: This is not totally true, UMD build is still available to use. Either by using script tag or as you can see in the solution to this issue by targeting the UMD build.

https://github.com/chartjs/Chart.js/issues/10875

So maby word is something like chartjs has become esm package by default

Author: done

* The order of the `ChartMeta` parameters have been changed from `<Element, DatasetElement, Type>` to `<Type, Element, DatasetElement>`.

### General

> * Chart.js becomes [ESM-only package](https://nodejs.org/api/esm.html). To use it in your project, it also should be ESM:

Reviewer 4: ```suggestion
* Chart.js becomes a [ESM-only package](https://nodejs.org/api/esm.html). To use it in your project, it should support ESM as well:
```

> * Chart.js becomes [ESM-only package](https://nodejs.org/api/esm.html). To use it in your project, it also should be ESM:
>   ```json
>   // package.json
>   {
>     "type": "module"
>   }
>   ```
>   If you are experiencing problems with Jest, you should follow [this doc](https://jestjs.io/docs/ecmascript-modules) to enable ESM support. Or, we can recommend you migrate to [Vitest](https://vitest.dev/). Vitest has ESM support out of the box and [has almost the same API as Jest](https://vitest.dev/guide/migration.html#migrating-from-jest). [Here is our example of migration](https://github.com/reactchartjs/react-chartjs-2/commit/7f3ec96101d21e43cae8cbfe5e09a46a17cff1ef).

Reviewer 4: ```suggestion
* Chart.js becomes a [ESM-only package](https://nodejs.org/api/esm.html). To use Chart.js, your application should support ESM as well. Make sure to have this in your `package.json`:
  ```json
  {
    "type": "module"
  }
  ```
  If you are experiencing problems with [Jest](https://jestjs.io), follow its [documentation](https://jestjs.io/docs/ecmascript-modules) to enable the ESM support. Or, we can recommend you migrating to [Vitest](https://vitest.dev/). Vitest has the ESM support out of the box and [almost the same API as Jest](https://vitest.dev/guide/migration.html#migrating-from-jest). See an [example of migration](https://github.com/reactchartjs/react-chartjs-2/commit/7f3ec96101d21e43cae8cbfe5e09a46a17cff1ef).
```

Author: Project should be ES module itself.

> * Chart.js becomes [ESM-only package](https://nodejs.org/api/esm.html). To use it in your project, it also should be ESM:
>   ```json
>   // package.json
>   {
>     "type": "module"
>   }
>   ```
>   If you are experiencing problems with Jest, you should follow [this doc](https://jestjs.io/docs/ecmascript-modules) to enable ESM support. Or, we can recommend you migrate to [Vitest](https://vitest.dev/). Vitest has ESM support out of the box and [has almost the same API as Jest](https://vitest.dev/guide/migration.html#migrating-from-jest). [Here is our example of migration](https://github.com/reactchartjs/react-chartjs-2/commit/7f3ec96101d21e43cae8cbfe5e09a46a17cff1ef).

Reviewer 4: ```suggestion
* Chart.js becomes a [ESM-only package](https://nodejs.org/api/esm.html). To use Chart.js, your application should also be a ES module. Make sure to have this in your `package.json`:
  ```json
  {
    "type": "module"
  }
  ```
  If you are experiencing problems with [Jest](https://jestjs.io), follow its [documentation](https://jestjs.io/docs/ecmascript-modules) to enable the ESM support. Or, we can recommend you migrating to [Vitest](https://vitest.dev/). Vitest has the ESM support out of the box and [almost the same API as Jest](https://vitest.dev/guide/migration.html#migrating-from-jest). See an [example of migration](https://github.com/reactchartjs/react-chartjs-2/commit/7f3ec96101d21e43cae8cbfe5e09a46a17cff1ef).
```
